### PR TITLE
Move CI from old Ubuntu images to Debian

### PR DESCRIPTION
**Pull request by cajun-rat**
Aktualizr CI builds are running on a very old version of Ubuntu. Bionic was released in April 2018 and ended support May 2023. Xenial was released in April 2016 and while it does have 10 years of support until 2026 it is stil ancient.

Instead use Debian (old)stable for the main test image. At the moment we don't build cleanly on Debian Stable (Bookworm). I've got a set of changes locally to fix that, but the first job is to get the CI clean, then we can look at moving it forward.

Finally, this nukes the old codecov stuff. No-one ever looked at those results, and if we want something that is low-cost to maintain we need to simplify where possible.

### CODE REVIEW

File: scripts/publish_github_rls.py
Comment (rborn-tx) [inline scripts/publish_github_rls.py:1]: @cajun-rat Similar question here: Is the idea to no longer publish release tarballs?

**Review by rborn-tx — APPROVED** [review]
I raised a couple of questions just for my understanding, but regardless of the answers changes look good to me.

File: scripts/publish_github_docs.sh
[the remark this answers is not in the document]
Comment (cajun-rat) [reply scripts/publish_github_docs.sh:1]: The docs are old and just API docs. I think it is better to nuke this for now to save the maintenance work and we can resurrect it if we want it in the future

File: scripts/publish_github_rls.py
Comment (cajun-rat) [reply scripts/publish_github_rls.py:1]: Yep. I don't think anyone uses them. The meta-updater layer pulls the code in from git, which makes a whole lot more sense

**Pull request merged**